src/Template/Element/sentences/list_form.ctp

```diff
@@ -0,0 +1,43 @@
+<?php
+use App\Model\CurrentUser;
+use Cake\ORM\TableRegistry;
+
+$SentencesLists = TableRegistry::getTableLocator()->get('SentencesLists');
+$lists = $SentencesLists->getUserChoices(
+    CurrentUser::get('id'), $sentenceId, true
+);
+$mostRecentList = $this->request->getSession()->read('most_recent_list');
+$listsJSON = htmlspecialchars(json_encode($lists), ENT_QUOTES, 'UTF-8');
```

Comment:
I believe this is temporary? It’s not a good idea, for two reasons:
1. The variable should be passed to the view instead.
2. The page is going to be cluttered with that same json data duplicated over and over in every sentence block.

We already have this problem for the autocomplete language list that is duplicated once, I think it’s a good time to find an alternative way to do this. Maybe [something like this](https://stackoverflow.com/questions/28402199/how-to-initiate-global-data-in-angularjs)?

Reply:
It was temporary, yes.

I made an attempt at a solution using a `.factory()` to store the data and initializing it from `app.run()`. At the same time it deals with the 1st point I mentioned in the "Improvements for another iteration" :)

Reply:
That's great!